src/shared_realm.hpp

```diff
@@ -295,8 +295,13 @@ class Realm : public std::enable_shared_from_this<Realm> {
 
     uint64_t m_schema_version;
     Schema m_schema;
+    util::Optional<Schema> m_new_schema;
     uint64_t m_schema_transaction_version = -1;
 
+    // FIXME: this should be a Dynamic schema mode instead, but only once
```

Comment:
Dynamic realms currently don't work, so at the minimum this shouldn't make things any worse.